tests/cert.rs

```diff
@@ -0,0 +1,225 @@
+#[macro_use]
+extern crate tracing;
+
+mod common;
+mod vault_bind_mounts_container;
+
+use std::collections::HashMap;
+use std::fs;
+
+use dockertest_server::Test;
+use rcgen::{BasicConstraints, Certificate, CertificateParams, IsCa};
+use tempfile::TempDir;
+use test_log::test;
+use vault_bind_mounts_container::{VaultServer, VaultServerConfig};
+use vaultrs::auth::cert::{self};
+use vaultrs::client::{Client, VaultClient, VaultClientSettingsBuilder};
+use vaultrs::error::ClientError;
+use vaultrs::sys::auth;
+
+use crate::common::{PORT, VERSION};
+
+#[cfg(not(feature = "native-tls"))]
+#[test]
+fn test() {
+    let certs = generate_certs();
+    let test = new_tls_test(&certs.serialized_cert_dir);
+    test.run(|instance| async move {
+        let server: VaultServer = instance.server();
+
+        let ca_cert_path = certs
+            .serialized_cert_dir
+            .path()
+            .to_path_buf()
+            .join("ca_cert.pem")
+            .to_str()
+            .unwrap()
+            .to_string();
+        let client_cert_str = certs
+            .client_cert
+            .serialize_pem_with_signer(&certs.ca_cert)
+            .unwrap();
+        let mut data = client_cert_str.as_bytes().to_vec();
+        let mut data2 = certs
+            .client_cert
+            .serialize_private_key_pem()
+            .as_bytes()
+            .to_vec();
+        data.append(&mut data2);
+        let identity = reqwest::Identity::from_pem(&data).unwrap();
+
+        let client = VaultClient::new(
+            VaultClientSettingsBuilder::default()
+                .address(format!("https://localhost:{PORT}"))
+                .token(server.token.clone())
+                .ca_certs(vec![ca_cert_path])
+                .identity(Some(identity))
+                .build()
+                .unwrap(),
+        )
+        .unwrap();
```

Comment:
This Handles the error if it is the expected `incompatible TLS identity type` and if the feature `native-tls` true and returns the function.
```suggestion
let client = match VaultClient::new(
            VaultClientSettingsBuilder::default()
                .address(format!("https://localhost:{PORT}"))
                .token(server.token.clone())
                .ca_certs(vec![ca_cert_path])
                .identity(Some(identity))
                .build()
                .unwrap(),
        ) {
            Ok(c) => c,
            Err(err) => {
                assert!(err
                    .source()
                    .unwrap()
                    .source()
                    .unwrap()
                    .to_string()
                    .eq("incompatible TLS identity type"));
                assert!(cfg!(feature = "native-tls").eq(&true));
                return;
            }
        };
```